sphinx/theming.py

```diff
@@ -286,3 +289,50 @@ def create(self, name):
                                    '(missing theme.conf?)') % name)
 
         return Theme(name, self.themes[name], factory=self)
+
+
+def generator_main(argv=sys.argv):
+    # type: (List[str]) -> int
+    from sphinx import __display_version__, package_dir
+
+    USAGE = ('Sphinx v%s\n'
+             'Usage: %%prog [options] [name]\n' % __display_version__)
+
+    parser = optparse.OptionParser(USAGE, version='Sphinx v%s' % __display_version__)
```

Comment:
Let's use argparse
There is an issue that replacing optparse with argparse: #3259.
IMO, it seems unnecessary to use optparse with newly introduced code.
(It's ok to release b1 as it is and replace it with argparse in b2)